src/seals/base_envs.py

```diff
@@ -151,51 +138,50 @@ def __init__(self, env: ResettablePOMDP[State, Observation, Action]) -> None:
         super().__init__(env)
         self._observation_space = env.state_space
 
-    def reset(self) -> State:
+    def reset(
+        self, seed: Union[int, None] = None, options: Union[Dict[str, Any], None]= None
+    ) -> Tuple[StateType, Dict[str, Any]]:
         """Reset environment and return initial state."""
-        self.env.reset()
-        return self.env.state
+        _, info = self.env.reset(seed=seed, options=options)
+        return self.env.state, info
 
-    def step(self, action) -> Tuple[State, float, bool, dict]:
+    def step(self, action) -> Tuple[StateType, float, bool, bool, dict]:
         """Transition state using given action."""
-        obs, reward, done, info = self.env.step(action)
-        return self.env.state, reward, done, info
+        _, reward, terminated, truncated, info = self.env.step(action)
+        return self.env.state, reward, terminated, truncated, info
 
 
 class ResettableMDP(
-    ResettablePOMDP[State, State, Action],
+    ResettablePOMDP[StateType, StateType, ActType],
     abc.ABC,
-    Generic[State, Action],
+    Generic[StateType, ActType],
 ):
     """ABC for MDPs that are resettable."""
 
-    def __init__(
-        self,
-        *,
-        state_space: gym.Space,
-        action_space: gym.Space,
-    ):
-        """Build resettable MDP.
+    @property
+    def observation_space(self) -> spaces.Space[StateType]:
+        """Observation space."""
+        return self.state_space
 
-        Args:
-            state_space: gym.Space containing possible states.
-            action_space: gym.Space containing possible actions.
-        """
-        super().__init__(
-            state_space=state_space,
-            observation_space=state_space,
-            action_space=action_space,
-        )
+    @observation_space.setter
```

Comment:
When would you want to change the observation space on the fly rather than at construction?

Reply:
You are right. This should not be needed.